app/src/main/java/it/niedermann/owncloud/notes/persistence/SyncWorker.java

```diff
@@ -44,7 +43,14 @@ public Result doWork() {
         return Result.success();
     }
 
-    public static void update(@NonNull Context context, Boolean backgroundSync) {
+    /**
+     * Set up sync work to enabled every 15 minutes or just disabled
+     * https://github.com/stefan-niedermann/nextcloud-notes/issues/1168
+     * @param context the application
+     * @param backgroundSync the toggle result backgroundSync
+     */
+
+    public static void update(@NonNull Context context, @NonNull Boolean backgroundSync) {
```

Comment:
Instead of a `@NonNull Boolean`, use just a `boolean` which is basically the same 😉

Reply:
Yes, they are the same. I have a question that are there some special cases causes them to be different?

Reply:
The difference is, that a `@NonNull Boolean` **can** be `null`, it is just not *supposed* to be `null`. IDEs like Android Studio will be able to display warnings if you pass `null` into this method, but it will not fail if you actually do it. If you use a value from a unknown source, the annotation will not be able to ensure that there is no `null` passed, while the application directly will fail early at this moment, because the implicit unboxing will throw an exception in this case.

So for primitive values, if you want to make sure, that there is no `null`, go for the primitive type. For all other object types and `String`, go for the annotation.

Other languages like kotlin distinguish between `Boolean` and `Boolean?` as type and provide null safety on compile time, while the `@NonNull` and `@Nullable` annotations in java will not provider compile time "safety" but will support developers well, when they are used across the application. They add more semantic to the wanted type and make the assumptions of the developer more clear.

Reply:
Ok, thanks.